src/main/java/uk/gov/hmcts/divorce/noticeofchange/event/SystemApplyNoticeOfChange.java

```diff
@@ -103,6 +105,8 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToStart(final CaseDeta
 
         CaseData responseData = objectMapper.convertValue(data, CaseData.class);
 
+        caseFlagsService.resetSolicitorCaseFlags(responseData, isApplicant1);
```

Comment:
I would shift it above the API call that's done via `assignCaseAccessClient`, where we make the other data changes (e.g buildChangeOfRepresentative and resetConditionalOrderFields). 

If it's arranged in that order and there's some error in the code, the event will fail, case data will roll back to what it was originally, and the sol won't have been granted case access through AAC API